test/FS.Embedded.Tests/Manifest/ManifestParserTests.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.FileProviders.Embedded.Tests;
+using Xunit;
+
+namespace Microsoft.Extensions.FileProviders.Embedded.Manifest
+{
+    public class ManifestParserTests
+    {
+        [Fact]
+        public void Parse_UsesDefaultManifestNameForManifest()
+        {
+            // Arrange
+            var assembly = new TestAssembly(
+                TestEntry.Directory("unused",
+                    TestEntry.File("sample.txt")));
+
+            // Act
+            var manifest = ManifestParser.Parse(assembly);
+
+            // Assert
+            Assert.NotNull(manifest);
+        }
+
+        [Fact]
+        public void Parse_FindsManifestWithCustomName()
+        {
+            // Arrange
+            var assembly = new TestAssembly(
+                TestEntry.Directory("unused",
+                    TestEntry.File("sample.txt")),
+                manifestName: "Manifest.xml");
+
+            // Act
+            var manifest = ManifestParser.Parse(assembly, "Manifest.xml");
+
+            // Assert
+            Assert.NotNull(manifest);
+        }
+
+        [Fact]
+        public void Parse_ThrowsForEntriesWithDifferentCasing()
```

Comment:
Does the manifest generation task issue warnings or errors for this scenario? If it doesn't, consider adding this so we can catch this early

Reply:
I've thought of this but decided to leave it for now for a few reasons:
* I don't want the manifest to be opinionated.
* We do case-insensitivity by default because that prevents issues with running on windows and failing on Linux.
* Its not very common to have files in your project (content) that only differ on casing.
* We might need to support case sensitivity in the future for things like static files once we have a clearer picture of how this will be used in the context of razor in class libraries. (We explicitly excluded providing this option for preview, but it might make a come back.)